javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/VirtualFieldDetector.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.bootstrap;
+
+import io.opentelemetry.instrumentation.api.cache.Cache;
+import java.util.Arrays;
+
+/** Helper class for detecting whether given class has virtual fields. */
+public final class VirtualFieldDetector {
+
+  private static final Cache<Class<?>, Boolean> classesWithVirtualFields = Cache.weak();
```

Comment:
ClassValue?

Reply:
As ClassValue is not mutable, it works best if you can compute the value on first use. Here we can't really compute the value, so the best we could do would be to use a mutable object as value of ClassValue, initialize it to false, and update it when the actual value is known. With current solution we can ignore the classes that don't have virtual fields, with ClassValue we would probably end up initializing a ClassValue for these too.

Reply:
makes sense, thx!